src/components/photo/PhotoList.js

```diff
@@ -0,0 +1,87 @@
+import React from "react";
+import { useLocation, useNavigate } from "react-router-dom";
+import { useQuery } from "react-query";
+import { FcPlus } from "react-icons/fc";
+import styled, { ThemeProvider } from "styled-components";
+import axios from "axios";
+
+import theme from "../../styles/theme";
+import { ERROR_MESSAGE } from "../../constants";
+import ResponseMessage from "../common/ResponseMessage";
+import Photo from "./Photo";
+
+const PhotoList = () => {
+  const { search: query } = useLocation();
+  const navigate = useNavigate();
+
+  const handlePhotoClick = event => {
+    const photoId = event.currentTarget.id;
+
+    navigate(`../${photoId}`);
+  };
+
+  const handleNewButtonClick = () => {
+    navigate(`../options${query}`);
+  };
+
+  const getPhotos = query => {
+    return axios.get(`/point/photos${decodeURI(query)}`);
+  };
+
+  const { data, isLoading, isFetching, isError, error } = useQuery(
+    "photos",
+    () => getPhotos(query),
+    {
+      select: response => response.data,
+    }
+  );
+
+  if (isLoading || isFetching) {
+    return <ResponseMessage message="사진을 불러오는 중입니다..." />;
+  }
+
+  if (data?.error) {
+    if (data.error.code === 400) {
+      return <ResponseMessage message={ERROR_MESSAGE.BAD_REQUEST} />;
+    }
+
+    if (data.error.code === 500) {
+      return <ResponseMessage message={ERROR_MESSAGE.SERVER_UNSTABLE} />;
+    }
+  }
+
+  return isError ? (
+    <ResponseMessage message={error.message} />
+  ) : (
+    <ThemeProvider theme={theme}>
+      <Container>
+        {data?.photos.map(photo => (
+          <Photo key={photo.id} photo={photo} onClick={handlePhotoClick} />
```

Comment:
```suggestion
          <Photo key={photo.id} photoInfo={photo} onPhotoClick={handlePhotoClick} />
```

`onclick` 보다는 `onPhotoClick` / `photo` 보다는 `photoInfo` 처럼 조금 더 직관적인 네이밍이 좋을 것 같은데,
두 분 생각은 어떠신가요-??

Reply:
저는 개인적으로 어떤 액션이 일어났을때 실행되는지만 알려줄 수 있도록 prop 네이밍을 general 하게 지어주는 것이 좋다고 생각했어요! 컴포넌트 이름이 Photo 라서 이벤트핸들러의 네이밍이 handlePhotoClick 으로 명확하게 알려주고 있다고 생각하기도 해서 이렇게 넘겨주었는데요~!

저는 prop으로 넘겨줄 때 `photo` 라는 네이밍이 `photoInfo` 라는 네이밍이 별로 다르지 않은 느낌으로 받아들여지는 것 같은데 이 부분들에 대해서 서영님 의견도 궁금하네요~!ㅎㅎ